src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -423,13 +415,14 @@ static Type GetMemberType(MemberInfo memberInfo)
             // record types or anonymous types are used.
             // The property name key does not use [JsonPropertyName] or PropertyNamingPolicy since we only bind
             // the parameter name to the object property name and do not use the JSON version of the name here.
-            var nameLookup = new Dictionary<ParameterLookupKey, ParameterLookupValue>(PropertyCacheArray!.Length);
+            var nameLookup = new Dictionary<ParameterLookupKey, ParameterLookupValue>(PropertyCache!.Count);
 
-            foreach (JsonPropertyInfo jsonProperty in PropertyCacheArray!)
+            foreach (KeyValuePair<string, JsonPropertyInfo?> kvp in PropertyCache.List)
```

Comment:
Do we care that using the list indexer is faster?

Reply:
Since the enumerator and KeyValuePair are value types, there wasn't a measurable impact on perf.